desk/ted/group/create.hoon

```diff
@@ -0,0 +1,110 @@
+::  create and preset a group
+::
+::  run via web at
+::  https://ship/spider/group/groups/group-create-thread/group-create/group-create/group-ui-1
+::
+/-  spider
+/-  g=groups, gt=groups-thread, c=channels, meta
+/+  io=strandio
+::  keep warm
+/=  in-mark  /mar/group/create-thread
+::
+=,  strand=strand:spider
+^-  thread:spider
+::
+|=  arg=vase
+=/  m  (strand ,vase)
+=*  n  (strand ,~)
+^-  form:m
+=+  !<(arg=(unit create-group:gt) arg)
+?>  ?=(^ arg)
+=*  create  u.arg
+=*  channels  channels.create
+;<  =bowl:spider  bind:m  get-bowl:io
+?>  =(p.group-id.create our.bowl)
+;<  exists=?  bind:m
+  (scry:io ? /gx/groups/exists/(scot %p p.group-id.create)/[q.group-id.create]/noun)
+~&  group+[group-id.create exists=exists]
+::  create the group if it does not exist
+::
+;<  ~  bind:m
+  ?:  exists  (pure:n ~)
+   =/  =create:g
+     =,  create
+     :*  q.group-id
+         title.meta
+         description.meta
+         image.meta
+         cover.meta
+         [%shut ~ ~]
+         %-  ~(gas by *(map ship (set sect:g)))
+         %+  turn  ~(tap in guest-list)
+         |=(=ship [ship ~])
+         ::
+         &
+     ==
+  (poke:io [our.bowl %groups] group-create+!>(`create:v6:g`create))
+::  set metadata and cordon if the group exists
+;<  ~  bind:m
+  ?.  exists  (pure:n ~)
+  ;<  ~  bind:n
+    =/  =action:v5:g
+      group-id.create^[now.bowl [%meta meta.create]]
+    (poke:io [our.bowl %groups] group-action-4+!>(action))
+  =/  =action:v5:g
+    :+  group-id.create
+      now.bowl
+    [%cordon %shut %add-ships %pending guest-list.create]
+  (poke:io [our.bowl %groups] group-action-4+!>(action))
```

Comment:
I thought groups did this for you in the create poke, no?

Reply:
The code below `?.`  runs only if the group had already existed – we don't send the create poke in that case.